public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -387,9 +387,13 @@ export const enhanceDataFrame = (dataFrame: DataFrame, config: LokiOptions | nul
  * Transform derivedField config into dataframe field with config that contains link.
  */
 function fieldFromDerivedFieldConfig(derivedFieldConfigs: DerivedFieldConfig[]): Field<any, ArrayVector> {
+  const dataSourceSrv = getDataSourceSrv();
+
   const dataLinks = derivedFieldConfigs.reduce((acc, derivedFieldConfig) => {
     // Having field.datasourceUid means it is an internal link.
     if (derivedFieldConfig.datasourceUid) {
+      const dsSettings = dataSourceSrv.getDataSourceSettingsByUid(derivedFieldConfig.datasourceUid);
```

Comment:
Guess this won't be necessary at some point?

Reply:
I think we will likely always have to lookup the data source to set a link title at least. Or we let the user set the link title when they define the derived field

Reply:
OK I see thanks!